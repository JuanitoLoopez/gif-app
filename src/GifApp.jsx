import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GifApp = () => {

  const [categories, setCategories] = useState([
    'One Punch',
    'Dragon Ball'
  ])

  const onAddCategory = (newCategory) => {
    if( categories.includes(newCategory) ) return;

    setCategories( [newCategory, ...categories] )
  }

  return (
    <>
      <h1>GifApp</h1>

      <AddCategory onNewCategory={onAddCategory}/>

      <button onClick={ onAddCategory }>Agregar</button>
      
      { 
        categories.map(category =>
          (
            <GifGrid key={category} category={category} />
          )
        )
      }

    </>
  )
}

