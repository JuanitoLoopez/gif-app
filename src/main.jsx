import React from 'react'
import ReactDOM from 'react-dom/client'
import { GifApp } from './GifApp'
import './styles.css';

// const apiKey = IXQ3lZBrrknbCt5NB3faPvl5aClrGv2w
ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifApp />
  </React.StrictMode>,
)
