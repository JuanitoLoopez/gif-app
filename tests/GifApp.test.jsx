import { render, screen, fireEvent } from "@testing-library/react"
import { GifApp } from "../src/GifApp"

describe('GifApp testing', () => {

    const category = 'Vegeta'

    test('should match with snapshot', () => {

        const {container} = render(<GifApp />)
        expect(container).toMatchSnapshot()

    })

    test('should not execute onAddCategory if the input has a value', () => {
        
        const onAddCategory = jest.fn()
        
        render(<GifApp />)
        const input = screen.getByRole('textbox')
        const form = screen.getByRole('form')
        
        fireEvent.input(input, { target: {value: category} })
        fireEvent.submit(form)
        
        expect(input.value).toBe('')
        expect(onAddCategory).not.toHaveBeenCalled()
        expect(onAddCategory).toHaveBeenCalledTimes(0)
        
    })
    
    // test('should execute onAddCategory if the input has a value', () => {
        
        // const onAddCategory = jest.fn()
        // render(<GifApp />)
    
        // const form = screen.getByRole('form')
        // fireEvent.submit(form)
    
        // expect(input.value).toBe('')
        // expect(onNewCategory).toHaveBeenCalled()
        // expect(onNewCategory).toHaveBeenCalledTimes(1)
        // expect(onAddCategory).toHaveBeenCalledWith(category) 

    // })

})