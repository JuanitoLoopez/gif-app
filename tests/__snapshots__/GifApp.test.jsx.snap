// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifApp testing should match with snapshot 1`] = `
<div>
  <h1>
    GifApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search Gif"
      type="text"
      value=""
    />
  </form>
  <button>
    Agregar
  </button>
  <h3>
    One Punch
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    Dragon Ball
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
